### PR TITLE
feat: first draft for accept invites

### LINKED ISSUE #180: Add `AcceptInvitation` method to organization-svc
This method should allow an authenticated user to move an invitation from `Pending` to `Accepted`. The user should also be a part of the organization afterwards.

_Previously #179_

### PR DESCRIPTION
**Pull request by florian-sabonchi**
closes #180

### CODE REVIEW

**Review by use-to — CHANGES_REQUESTED** [review]
Rename to `AcceptInvitation`

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
> This method should allow an authenticated user to move an invitation from Pending to Accepted. The user should also be a part of the organization afterwards.

- [ ] Only allow the acceptance of an invitation when the current state is pending. When the invitation is already accepted, return a success to be idempotent.
- [ ] Add user to the organization
- [ ] Check if the email of the invitation equals the current user

File: services/user-svc/internal/organization/organization.go [hunk 1]
```diff
@@ -209,17 +203,98 @@ func (s ServiceServer) InviteMember(ctx context.Context, req *pb.InviteMemberReq
...
+	if err := db.Model(&invitation).Update("state", Accepted).Error; err != nil {
 		log.Warn().Err(err).Msg("database error")
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	log.Info().
-		Str("email", req.Email). // TODO: Revisited for privacy reasons
-		Str("organizationId", organizationId.String()).
-		Msg("user invited to organization")
+	userID, err := uuid.Parse(claims.Sub)
```
Comment (MaxSchaefer) [inline services/user-svc/internal/organization/organization.go:281]:
```suggestion
	userID, err := common.GetUserID(ctx)
```

File: services/user-svc/internal/organization/organization.go [hunk 2]
```diff
@@ -209,17 +203,98 @@ func (s ServiceServer) InviteMember(ctx context.Context, req *pb.InviteMemberReq
 		}
 	}
 
-	if err := db.Create(&invite).Error; err != nil {
+	// check if already an invitation exists
+	invite := Invitation{}
+
+	if err := db.Where("email = ? AND organization_id = ?", req.Email, organizationId).First(&invite).Error; err != nil {
```
Comment (MaxSchaefer) [inline services/user-svc/internal/organization/organization.go:209]: Thank you for thinking of it. Beside email and organization, we should also check if the invitation is in progress. That means, only create a new invitation if we not find one with `state == Pending` or `state == Accepted`.

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: services/user-svc/internal/organization/organization.go [hunk 3]
```diff
@@ -209,17 +203,98 @@ func (s ServiceServer) InviteMember(ctx context.Context, req *pb.InviteMemberReq
...
+	// Add user to organization
+	member := Member{
+		UserID:         userID,
+		OrganizationID: currentInvitation.OrganizationID,
+	}
+
+	if err := db.Create(&member).Error; err != nil {
+		log.Warn().Err(err).Msg("database error")
+		return nil, status.Error(codes.Internal, err.Error())
+	}
```
[the remark this answers is not in the document]
Comment (florian-sabonchi) [reply services/user-svc/internal/organization/organization.go:295]: Sorry, not seen thanks

**Review by MaxSchaefer — APPROVED** [review]
Good job 🙌 (squash please)

**Comment by MaxSchaefer** [comment]
@use-to pending requested changes

**Review by use-to — APPROVED** [review]

**Pull request merged**